blockstack_client/config.py

```diff
@@ -987,62 +994,64 @@ def get_app_config(app_name, path=CONFIG_PATH):
 
 
 def update_config(section, option, value, config_path=CONFIG_PATH):
-
     parser = SafeConfigParser()
 
     try:
         parser.read(config_path)
-    except Exception, e:
+    except Exception as e:
         log.exception(e)
         return None
 
-    if parser.has_option(section, option):
-        parser.set(section, option, value)
+    if not parser.has_option(section, option):
+        return
+
+    parser.set(section, option, value)
+
+    with open(config_path, 'wb') as configfile:
+        parser.write(configfile)
+
 
-        with open(config_path, 'wb') as configfile:
-            parser.write(configfile)
+def get_version_parts(whole, func):
+    return [func(_.strip()) for _ in whole[0:3]]
 
 
-def semver_match( v1, v2 ):
+def semver_match(v1, v2):
     """
     Verify that two semantic version strings match:
     the major, minor, and patch versions must be equal.
     """
-    v1_parts = v1.split(".")
-    v2_parts = v2.split(".")
+    v1_parts = v1.split('.')
+    v2_parts = v2.split('.')
     if len(v1_parts) < 3 or len(v2_parts) < 3:
-        # one isn't a semantic version 
+        # one isn't a semantic version
         return False
 
-    v1_major, v1_minor, v1_patch = v1_parts[0].strip(), v1_parts[1].strip(), v1_parts[2].strip()
-    v2_major, v2_minor, v2_patch = v2_parts[0].strip(), v2_parts[1].strip(), v2_parts[2].strip()
-    if v1_major != v2_major:
-        return False
+    v1_major, v1_minor, v1_patch = get_version_parts(v1_parts, str)
+    v2_major, v2_minor, v2_patch = get_version_parts(v2_parts, str)
 
-    if v1_minor != v2_minor:
-        return False
+    # TODO: check why patch versions are not compared. see function docstring.
```

Comment:
Good catch!  The doc string is wrong; we only care about the major and minor versions.